src/Pact/Types/Term.hs

```diff
@@ -341,6 +414,11 @@ data Term n =
     , _tTableType :: !(Type (Term n))
     , _tMeta :: !Meta
     , _tInfo :: !Info
+    } |
+    TImplements {
+      _tInterfaceName :: !ModuleName
+    , _tModuleName :: !ModuleName
```

Comment:
Are we sure we need to track the declaring module name for an `implements`?

Reply:
My thinking was that we should, becase we want this declared within a module